### PR TITLE
GAN tweaks and Finalizations

### PR DESCRIPTION
**Pull request by gallupBenRyan**
Introduced many tweaks, bug fixes, and improvements in the GAN implementation. Progress is now tracked for every epoch with output model checkpoints, sample generated queries, and running generator/discriminator loss histories.

### CODE REVIEW

File: src/gan/gen_pyt/asdl/lang/eng/eng_asdl_helper.py [hunk 1]
```diff
@@ -65,11 +79,32 @@ def asdl_ast_to_english(asdl_ast_node):
...
             field_value = field.value
 
         tokens.append(field_value)
 
-    return ' '.join(tokens)
\ No newline at end of file
+    return ' '.join([x if x else '<None>' for x in tokens])
+
+
+# def unroll(asdl_ast_node):
```
Comment (gallupBenRyan) [inline src/gan/gen_pyt/asdl/lang/eng/eng_asdl_helper.py]: NB: I am already planning on removing this commented code.

File: src/gan/config.cfg [hunk 2]
```diff
@@ -1,38 +1,40 @@
 [GAN]
 ROLLOUT_UPDATE_RATE = 0.8
-ROLLOUT_NUM = 16
```
Comment (mhoover-gallup) [inline src/gan/config.cfg]: Are these stable more or less? Or likely to be tweaked over time? If they are going to be tweaked over time, it would probably make sense to pass some of these as command line arguments using `**kwargs` otherwise, there will be a lot of "changing config parameters and oops, do I need to commit that?"

File: src/gan/config.cfg [hunk 3]
```diff
@@ -1,38 +1,40 @@
...
-GENERATED_NUM = 100
+TOTAL_EPOCHS = 200
+GENERATED_NUM = 1000
 VOCAB_SIZE = 10
-SEQUENCE_LEN = 20
+SEQUENCE_LEN = 10
 
 data_dir = data
 glove_dir = ../../models
 glove_file = DA_glove_embeddings_300.pkl
```
Comment (mhoover-gallup) [inline src/gan/config.cfg]: Just be careful -- this isn't likely to change, but if there is no way to pass this is as a run-time argument, then it's possible a lot of work you think is using a different glove file is not.

File: src/gan/gen_pyt/model/lstm.py [hunk 4]
```diff
@@ -7,96 +7,12 @@
 import torch.nn.utils
 from torch.autograd import Variable
 from torch.nn import Parameter, init
-from torch.nn._functions.rnn import variable_recurrent_factory, StackedRNN
+# from torch.nn._functions.rnn import variable_recurrent_factory, StackedRNN
```
Comment (mhoover-gallup) [inline src/gan/gen_pyt/model/lstm.py]: Delete this and the one below?

File: src/gan/gen_test.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import argparse
```
Comment (mhoover-gallup) [inline src/gan/gen_test.py]: Based on the name here, what's going on with this? Is it just a short-hand test file or what? I guess what I'm asking is, how does this fit into the overall _production_ system or is it separate from it? If separate, then I'd suggest creating a folder called `tests` in `src/gan` and put this file there.

File: src/gan/rollout.py [hunk 6]
```diff
@@ -1,54 +1,140 @@
...
-    def update_params(self):
-        dct = {}
-        for name, param in self.ori_net.named_parameters():
-            dct[name] = param.data
-        for name, param in self.new_net.named_parameters():
-            if name.startswith('emb'):
-                param.data = dct[name]
-            else:
-                param.data = self.update_rate * param.data + (1 - self.update_rate) * dct[name]
+    # def update_params(self):
```
Comment (mhoover-gallup) [inline src/gan/rollout.py]: Delete? Uncomment?
